oggm/sandbox/run_alaska.py

```diff
@@ -127,65 +130,237 @@
 # realistic values. This might be undersirable for tidewater glaciers, and
 # it is possible to add special conditions for tidewater glaciers in the
 # function. For example: if gdir.is_tidewater: etc.
+# FOR THIS TEST THE FUNCTION HAS BEEN COMMENTED !!! FOR NOW
 
-gdir = gdirs[-1]
-
-# Test that frontal H is close to zero
+gdir = gdirs[3]
+#
+## Test that frontal H is close to zero
 cl = gdir.read_pickle('inversion_output', div_id=1)[-1]
 width = cl['width'][-1]
 thick = cl['volume'][-1] / cl['dx'] / width
+print('For the glacier', gdir.name)
 print('Without calving the width is {} m and the thick is {} m'.format(width, thick))
 
-# Try many C's (they should make sense in comparison to the size of the glacier)
-c_candidates = np.linspace(0, 0.1, 10)
-for c in c_candidates:
-
-    # First guess calving
-    gdir.inversion_calving_rate = c
-
-    # Recompute mu
-    tasks.distribute_t_stars([gdir])
-
-    # Inversion
-    tasks.prepare_for_inversion(gdir)
-    tasks.volume_inversion(gdir)
-
-    # We read the output, last pixel of the inversion
+# Defining a calving function
+def calving_from_depth(gdir):
+    """ Finds a calving flux based on the approaches proposed by
+        Huss and Hock, (2015) and Oerlemans and Nick (2005)
+
+    We take the initial output of the model and surface elevation data
+    to calculate the water depth of the calving front.
+    The same criteria of the 10th lowest percentile is used to find the
+    elevation at the terminus.
+    """
+    # Opens all the files we need
+    geom = gdir.read_pickle('geometries', div_id=1)
+    grids_file = gdir.get_filepath('gridded_data', div_id=1)
+    with netCDF4.Dataset(grids_file) as nc:
+        # Variables
+        glacier_mask = nc.variables['glacier_mask'][:]
+        glacier_ext = nc.variables['glacier_ext'][:]
+        topo = nc.variables['topo_smoothed'][:]
+        poly_pix = geom['polygon_pix']
+
+    # Finding the outline altitudes
+    x, y = tuple2int(poly_pix.exterior.xy)
+    ext_yx = tuple(reversed(poly_pix.exterior.xy))
+    zoutline = topo[y[:-1], x[:-1]]  # last point is first point
+
+    # Finds the altitude at the terminus in the same way as in the centerlines
+    # code
+    # Getting the lowest 10th percentile
+    perc = cfg.PARAMS['terminus_search_percentile']
+    deltah = cfg.PARAMS['terminus_search_altitude_range']
+
+    plow = np.percentile(zoutline, perc).astype(np.int64)
+
+    # the minimum altitude in the glacier
+    mini = np.min(zoutline)
+
+    # indices of where in the outline the altitude is lower than the qth
+    # percentile and lower than 100m higher, than the minimum altitude
+    ind = np.where((zoutline < plow) & (zoutline < (mini + deltah)))[0]
+
+    # We take the middle of this area
+    ind_term = ind[np.round(len(ind) / 2.).astype(np.int)]
+
+    # Getting the termminus altitude
+    t_altitude = zoutline[ind_term]
+
```

Comment:
Here, instead of recomputing what we already did in ``core.centerlines.compute_centerlines``, you could get the altitude from ``cl['hgt']`` I guess. You can check it first, and see if it is the same as what you compute.